android/WALT/app/src/main/java/org/chromium/latency/walt/MainActivity.java

```diff
@@ -91,6 +91,7 @@ protected void onResume() {
         final UsbDevice usbDevice;
         Intent intent = getIntent();
         if (intent != null && intent.getAction().equals(UsbManager.ACTION_USB_DEVICE_ATTACHED)) {
+            setIntent(null); // done with the intent
```

Comment:
getInent() and setIntent() interact with the last intent which caused the app to open. I'm getting a local reference to the intent and then clearing it from the system so that if onResume is triggered, but there is no new intent (such as when the screen is turned off and back on) it won't try to act based on the old intent.